### PR TITLE
custom hooks demo 

### PR DESCRIPTION
**Pull request by rohanzemoso**
# Demo for custom hooks in react typescript
## This project includes demo of use case of custom hooks, along with atomic design pattern, and material UI.

- screenshot:

image

### CODE REVIEW

File: customhooksreact/src/components/molecules/UserForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import InputField from '../../atoms/InputField/index';
+
+interface UserFormProps {
+  inputProps: {
+    value: string;
```
Comment (aritrazomoso) [inline customhooksreact/src/components/molecules/UserForm/index.tsx:6]: why object inside object

File: customhooksreact/src/hooks/useInput.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { useState } from 'react';
+
+const useInput = (initialValue: string) => {
```
Comment (aritrazomoso) [inline customhooksreact/src/hooks/useInput.ts:3]: use interface for return type

**Review by aritrazomoso — CHANGES_REQUESTED** [review]

File: customhooksreact/src/components/molecules/UserForm/index.tsx (hunk 1 above)
Comment (rohanzemoso) [reply customhooksreact/src/components/molecules/UserForm/index.tsx:6]: Using objects inside objects as props groups related properties together, improving code organization and readability, also simplifies prop management and forwarding to child components.

File: customhooksreact/src/hooks/useInput.ts (hunk 2 above)
Comment (rohanzemoso) [reply customhooksreact/src/hooks/useInput.ts]: used interface for return type, resolved.

**Review by aritrazomoso — APPROVED** [review]
looks good to me

**Pull request merged**